packages/frontend/src/utils/twoFactor/twoFactorBase.js

```diff
@@ -0,0 +1,214 @@
+                return perm.receiver_id === this.accountId &&
+                    perm.method_names.length === 4 &&
+                    perm.method_names.includes('add_request_and_confirm');
+            });
+    }
+
+    /**
+     * Returns `false` if multisig cannot be disabled by calling disable() directly
+     * i.e. account has too many LAKs to delete and re-create as FAKs for a single transaction
+     */
+    async isKeyConversionRequiredForDisable() {
+        const keys = await this.get2faLimitedAccessKeys();
+        return keys.length > LAK_DISABLE_THRESHOLD;
+    }
+
+    /**
+     * Converts multisig LAKs (excluding the active key in the wallet, provided here as the
+     * `signingPublicKey` argument) back to FAKs when there are too many key conversions to be
+     * executed within the same transaction.
+     *
+     * Note that this undermines the security provided by 2FA by converting existing multisig LAKs into FAKs
+     * capable of bypassing the 2FA prompt when signing transactions outside the wallet (transactions signed
+     * in the wallet will continue to require 2FA approval). This method should only be used to convert LAKs
+     * as a way to make 2FA disabling possible.
+     *
+     * To be deprecated after 2FA migration; this addresses a specific issue with multisig deployed via wallet.
+     * @param signingPublicKey the public key used to sign transactions in the wallet
+     */
+    async batchConvertKeysAndDisable(signingPublicKey) {
+        if (!signingPublicKey) {
+            throw new Error('the public key used to sign multisig transactions must be provided');
+        }
+
+        const { stateStatus } = await this.checkMultisigCodeAndStateStatus();
+        if (stateStatus !== 2 && stateStatus !== 1) {
+            throw new Error(`Can not deploy a contract to account ${this.accountId} on network ${this.connection.networkId}, the account state could not be verified.`);
+        }
+
+        let converted = 0;
+        const accessKeys = await this.get2faLimitedAccessKeys();
+        const keysToConvert = accessKeys
+            .filter(({ public_key }) => public_key !== signingPublicKey);
+
+        while (converted < (accessKeys.length - LAK_DISABLE_THRESHOLD)) {
+            const conversionActions = keysToConvert
+                .slice(converted, converted + LAK_CONVERSION_BATCH_SIZE)
+                .reduce((conversionActions, { public_key }) => {
+                    const { addKey, deleteKey, fullAccessKey } = nearApiJs.transactions;
+                    conversionActions.push(deleteKey(nearApiJs.utils.PublicKey.from(public_key)));
+                    conversionActions.push(addKey(nearApiJs.utils.PublicKey.from(public_key), fullAccessKey()));
+
+                    return conversionActions;
+                }, []);
+
+            await this.signAndSendTransaction({
+                receiverId: this.accountId,
+                actions: conversionActions
+            });
```

Comment:
I have tried to ran the code that has many FAKs with function call. It worked smoothly but it also took some time to get things cleared. So for UX improvement, wonder if we should show some form of a notification on right top corner showing:
50/200 Access keys are converted..
100/200 Access keys are converted..
150/200 Access keys are converted..
on when above `await this.sighANdSendTransaction` function call completes.

Reply:
Also in case RPC is down or due to some limitation, if `await this.signAndSendTransaction` call fails, is there a way to display error message and stop the while loop?

Reply:
> So for UX improvement, wonder if we should show some form of a notification on right top corner showing:

It's certainly an option for the design. We can track it on the instance easily enough without needing event emitters. However it's a one-time use feature for < 40 users who require more than one prompt so it'll come down to a matter of effort.

> Also in case RPC is down or due to some limitation, if `await this.signAndSendTransaction` call fails, is there a way to display error message and stop the while loop?

Once the retry/backoff logic reaches its limit, an exception would be thrown and surface to the user as usual. Next time they resume the disable workflow it would continue the remaining work.

Reply:
Thanks for clarification!